testiso - iscsi : misc fixes to improve reliability and disable the test for other arches than x86

jbtrystram opened the pull request:
- fix invalid After targets in iscsi services
- rename target container to targetd : Target is a systemd w
ord and seems to interfere with systemd, as running
systemctl stop target shuts the vm down
- add a wait for the targetd container to be ready before configuring the
iscsi target. this should be removed when we ship systemd 254 in rhcos
- Add a healthcheck to the target container and make sure we mark the
systemd service as started only when targetcli is ready to receive
commands. Note this is commented out due to an older systemd version in
rhcos
- update virtserial channel completion channel name to "testisocompletion"
to allow the coreos-test-entered-emergency-target.service to write to
it, as it expects this name.
This will allow correct failure detection rather than waiting for QEMU
timeout.
- add a restart policy on setup-targetcli as it may start too fast so
we allow it to restart before marking it failed. Disabled for now
waiting for a systemd update in rhcos

Disable the iscsi test on other arches than x86 : Nested virt does not work through cosa
ref https://github.com/coreos/fedora-coreos-tracker/issues/1657

jlebon commented on mantle/cmd/kola/resources/iscsi_butane_setup.yaml:
```diff
@@ -29,14 +29,15 @@ storage:
             WantedBy=multi-user.target
-            ExecStop=podman stop --time 1 target
+            [Service]
+            ExecStop=/usr/bin/podman stop --time 1 targetd
```
Hmm, we shouldn't need this (though does it work?). Let's see in https://github.com/coreos/fedora-coreos-config/pull/2816 if we can figure this out without resorting to it.

jlebon reviewed (commented):
Looks like this needs a rebase; the first two commits are already in.

The last two commits could be squashed together.

jbtrystram commented:
[console.txt](https://github.com/coreos/coreos-assembler/files/14029715/console.txt)

attached console.txt from aarch64 run of iso-offline-install-iscsi.bios
it was run with [this coreos-assembler changes](https://github.com/coreos/coreos-assembler/pull/3705#pullrequestreview-1839842072)

jlebon commented on mantle/cmd/kola/resources/iscsi_butane_setup.yaml:
```diff
@@ -33,7 +33,6 @@ storage:
             # Start by default on boot
             WantedBy=multi-user.target
             [Service]
```
Should remove this line too.

jlebon commented on mantle/cmd/kola/resources/iscsi_butane_setup.yaml:
```diff
@@ -129,6 +128,8 @@ systemd:
         Type=oneshot
         RemainAfterExit=yes
         ExecStart=/usr/local/bin/targetcli_script
+        Restart=on-failure
```
Do we still need this even with the healthcheck?

jbtrystram replied:
yes, I found that on aarch64 this script requires two runs to work , the first attempt fails. 
Maybe the healthcheck does not work the way I expect it to

jlebon reviewed (commented):
Can you clean up the commits? Needs a rebase and some of the commits squashed.

jbtrystram commented:
@jlebon I rebased the branch, squashed commits with a more detailed commit message and excluded the test from other arches than x86 with a note pointing to https://github.com/coreos/fedora-coreos-tracker/issues/1657

jlebon commented on mantle/cmd/kola/resources/iscsi_butane_setup.yaml:
```diff
@@ -35,7 +39,10 @@ storage:
-            podman exec target bash -exc "
+            # Sleep for a second in case targetcli is not ready
+            # FIXME: use RestartMode=direct instead in the systemd unit but we need systemd v254
+            sleep 1
```
I'm worried this could still flake on loaded CI systems. We could also do something like

```
while ! test -S /var/run/targetclid.sock; do sleep 1; done
```

inside the `podman exec` invocation.

jlebon reviewed (approved):
One suggestion, but OK as a follow-up too.

travier reviewed (approved):
Not tested but LGTM from a quick look.
